ranger-http/src/main/java/com/flipkart/ranger/http/servicefinder/HttpNodeDataSource.java

```diff
@@ -23,59 +41,66 @@
  */
 @Slf4j
 public class HttpNodeDataSource<T, D extends HTTPResponseDataDeserializer<T>> extends HttpNodeDataStoreConnector<T> implements NodeDataSource<T, D> {
+
+    private final Service service;
+
     public HttpNodeDataSource(
             Service service,
             final HttpClientConfig config,
             ObjectMapper mapper) {
-        super(service, config, mapper);
+        super(config, mapper);
+        this.service = service;
     }
 
-
     @Override
     public Optional<List<ServiceNode<T>>> refresh(D deserializer) {
```

Comment:
As mentioned somewhere before: Let's change this to return a list. We take care to return an empty list instead of null.

Reply:
Yeah. Done that across the codebase